interdiff_test.go

```diff
@@ -18,6 +19,20 @@ var interDiffFileTests = []struct {
 	{"test_examples/result_2.txt", "test_examples/diff_2_a.txt", "test_examples/diff_2_b.txt"},
 }
 
+var applyDiffFileTests = []struct {
+	sourceFile      string
+	diffFile        string
+	resultFile      string
+	finishedWithErr bool
+}{
+	{"test_examples/file_1.txt", "test_examples/diff_1_1_a.txt", "test_examples/file_1_a.txt", false},
+	{"test_examples/file_1.txt", "test_examples/diff_1_1_b.txt", "test_examples/file_1_b.txt", false},
+	{"test_examples/file_2.txt", "test_examples/diff_1_2_a.txt", "test_examples/file_2_a.txt", false},
+	{"test_examples/file_2.txt", "test_examples/diff_1_2_b.txt", "test_examples/file_2_b.txt", false},
+	// sourceFile and diffFile have different origin content.
+	{"test_examples/file_1.txt", "test_examples/diff_1_1_a_wrong_origin.txt", "test_examples/file_1_a.txt", true},
```

Comment:
```suggestion
	{"test_examples/file_1.txt", "test_examples/diff_1_a.txt", "test_examples/file_1_a.txt", false},
	{"test_examples/file_1.txt", "test_examples/diff_1_b.txt", "test_examples/file_1_b.txt", false},
	{"test_examples/file_2.txt", "test_examples/diff_2_a.txt", "test_examples/file_2_a.txt", false},
	{"test_examples/file_2.txt", "test_examples/diff_2_b.txt", "test_examples/file_2_b.txt", false},
	// sourceFile and diffFile have different origin content.
	{"test_examples/file_1.txt", "test_examples/diff_1_a_wrong_origin.txt", "test_examples/file_1_a.txt", true},
```

Did I miss something or can we simplify the file names a bit here?

Reply:
I will mark this files in next way.
For files:
`diff_f{i}_{c}.txt` -- `{c}` diff file for `file_{i}`.
`diff_f1_a.txt` -- `a` diff for `file_1.txt`.
`file_1.txt` + `diff_f1_a.txt` --> `file_1_a.txt`

For source (dir with files inside):
`diff_s{i}_{c}.txt` -- `{c}` diff file for `source_{i}`.
`diff_s1_a.txt` -- `a` diff for `source_1/`